crucible-llvm/src/Lang/Crucible/LLVM/MemModel.hs

```diff
@@ -710,10 +711,11 @@ doMallocSize sz bak allocType mut loc mem alignment = do
   let ptr   = LLVMPointer blk z
   let mem'  = mem{ memImplHeap = heap' }
   mem'' <- if laxLoadsAndStores ?memOpts
-                && mut == G.Mutable
                 && allocType == G.HeapAlloc
                 && indeterminateLoadBehavior ?memOpts == StableSymbolic
-           then doStoreStableSymbolic bak mem' ptr sz alignment
+           then if mut == G.Mutable
```

Comment:
A fair point. `doStoreStableSymbolic` would generate an additional assertion that the memory is mutable, but since we know whether the allocated memory is mutable or not from `doMalloc`'s argument, there's not much point in generating the assertion in the first place.

There are also two other call sites to `doStoreStableSymbolic` that warrant further scrutiny: one in `doAlloca` and one in `doInvalidate`. In `doAlloca`, we know that we are always allocating mutable memory, so I think `doConstStoreStableSymbolic` makes more sense there. In `doInvalidate`, we don't necessary know at a glance if the memory being invalidated is mutable or not, so I think it makes more sense to call `doStoreStableSymbolic` to generate the check-if-memory-is-mutable assertion.

In short, I've pushed a change that changes the uses of `doStoreStableSymbolic` in `do{Malloc,Alloca}` to `doConstStoreStableSymbolic` while leaving `doInvalidate` alone.